sentry-android-core/src/main/java/io/sentry/android/core/LifecycleWatcher.java

```diff
@@ -95,24 +101,30 @@ public void onStop(final @NotNull LifecycleOwner owner) {
   }
 
   private void scheduleEndSession() {
-    cancelTask();
-    timerTask =
-        new TimerTask() {
-          @Override
-          public void run() {
-            addSessionBreadcrumb("end");
-            hub.endSession();
-            runningSession.set(false);
-          }
-        };
-
-    timer.schedule(timerTask, sessionIntervalMillis);
+    synchronized (timerLock) {
+      cancelTask();
```

Comment:
is this lock reentrant? We're re-acquiring the lock in cancelTask. Well, it would deadklock otherwise